### ISSUE TITLE
`hoistTry` rule can make an error

### ISSUE THREAD

**Issue by ski-u** [nicklockwood/SwiftFormat#1374]
Hello, I tried to use `hoistTry` rule. It's a great idea!
But I found one case that `hoistTry` rule behaved as not expected.

We can reproduce it in the following way with `SwiftFormat v0.51.1`

1. We have this code, for example.
```swift
import XCTest

func test() throws {
    let newCount = Counter(
        count: 0
    )
    .addSomethingToCount(try XCTUnwrap(Int("1")))
}

struct Counter {
    let count: Int

    func addSomethingToCount(_ v: Int) -> Int {
        count + v
    }
}
```

2. And run `swiftformat .`
3. `SwiftFormat` update the code as following
```swift
import XCTest

func test() throws {
    let newCount = Counter(
        count: 0
    )
    try .addSomethingToCount(XCTUnwrap(Int("1")))  // <- Changed this line by SwiftFormat
}

struct Counter {
    let count: Int

    func addSomethingToCount(_ v: Int) -> Int {
        count + v
    }
}
```

4. Now we have an error like:
<img width="837" alt="スクリーンショット 2023-03-03 14 25 36" src="https://user-images.githubusercontent.com/37182704/222638593-dd6a3782-e162-4667-9517-be5807bea585.png">

Is there any way to avoid this besides disabling `hoistTry` rule?

---

### About my environment
```
❯ swift --version
swift-driver version: 1.62.15 Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
```
```
❯ xcodebuild -version
Xcode 14.2
Build version 14C18
```

**Comment by nicklockwood** [comment]
@ski-u thanks for reporting this. I've fixed it in the debug branch and will include the fix in the next release. In the meantime you can fix this case by manually hoisting the try to the correct location, which should prevent SwiftFormat mangling that line.

**Comment by ski-u** [comment]
@nicklockwood I see! Thank you for your work on this issue!

**Comment by nicklockwood** [comment]
@ski-u fix landed in 0.51.2
